test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+# coding: utf-8
+#from ptl.tests.pbs_testsuite import *
+
+
+from ptl.utils.pbs_testsuite import *
+import re
+class Test_Rrecord_with_resources_used(PBSTestSuite):
+    """
+    This test suite tests whether the 'R' record in accounting logs has information on resources_used
+    in the following scenarios.
+        a)The node the job was running on goes down and node_fail_requeue timeout is hit.
+        b)It is rerun using qrerun <job-id>.
+        c)It is rerun using qrerun -Wforce <job-id>.
+        d)mom is restarted without any options or with the '-r' option
+    """
+
+    def setUp(self):
+        PBSTestSuite.setUp(self)
+
+        if len(self.moms) != 2:
+            self.skipTest('test requires two MoMs as input, ' +
+                 'use -p moms=<mom1>:<mom2>')
+
+        self.server.set_op_mode(PTL_CLI)
+
+        # PBSTestSuite returns the moms passed in as parameters as dictionary
+        # of hostname and MoM object
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.momA.delete_vnode_defs()
+        self.momB.delete_vnode_defs()
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        a = {'resources_available.ncpus': 4}
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostA, expect=True)
+        self.assertTrue(rv)
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostB, expect=True)
+        self.assertTrue(rv)
+
+    def common(self, is_nonrerunnable, restart_mom):
+        """
+        Set node_fail_requeue=5 on server
+        """
+        a = {ATTR_nodefailrq: 5}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        """
+        Job script
+        """
+        test = []
+        test += ['#PBS -N RequeueTest\n']
+        test += ['#PBS -l ncpus=1\n']
+        test += ['echo Starting test at `date`\n']
+        test += ['sleep 1000\n']
+
+        test1 = []
+        test1 += ['#PBS -N RequeueTest\n']
+        test1 += ['#PBS -l nodes=1\n']
```

Comment:
Don't use -lnodes.  It's deprecated syntax.  Request -lselect=1:ncpus=1 -l place=scatter